src/safeds/data/tabular/containers/_table.py

```diff
@@ -578,7 +578,7 @@ def keep_only_columns(self, column_names: list[str]) -> Table:
         if len(invalid_columns) != 0:
             raise UnknownColumnNameError(invalid_columns)
         transformed_data = self._data[column_indices]
-        transformed_data.columns = [name for name in self._schema.get_column_names() if name in column_names]
+        transformed_data.columns = [self._schema.get_column_names()[i] for i in column_indices]
```

Comment:
```suggestion
        transformed_data.columns = column_names
```

Isn't this sufficient?

Reply:
yes you are right